varname/ignore.py

```diff
@@ -87,41 +87,53 @@ class IgnoreModule(IgnoreElem):
 
     def __init__(self, ignore: Union[ModuleType, str]) -> None:
         super().__init__(ignore)
-        self.is_module = isinstance(ignore, ModuleType)
-        if self.is_module:
-            modfile = getattr(self.ignore, '__file__', None)
-            if not modfile or not path.isfile(modfile):
-                setattr(self.ignore, MODULE_IGNORE_ID_NAME,
-                        f'<varname-ignore-{id(self.ignore)}>')
+        modfile = getattr(self.ignore, '__file__', None)
+        if not modfile or not path.isfile(modfile):
+            setattr(self.ignore, MODULE_IGNORE_ID_NAME,
+                    f'<varname-ignore-{id(self.ignore)}>')
 
     def match(self, frame_no: int, frameinfos: List[inspect.FrameInfo]) -> bool:
         frame = frameinfos[frame_no].frame
+        module = inspect.getmodule(frame)
+        if module:
+            return (module.__name__ == self.ignore.__name__ or
+                    module.__name__.startswith(f'{self.ignore.__name__}.'))
 
-        if self.is_module:
-            module = inspect.getmodule(frame)
-            if module:
-                return (module.__name__ == self.ignore.__name__ or
-                        module.__name__.startswith(f'{self.ignore.__name__}.'))
-
-            return (getattr(self.ignore, MODULE_IGNORE_ID_NAME, None) ==
-                    frame.f_globals.get(MODULE_IGNORE_ID_NAME, ''))
-
-        if path.isdir(self.ignore):
-            # ignore all calls from modules from a directory
-            # Used internally currently to ignore standard libraries
-            if not self.ignore.endswith('/'):
-                self.ignore = f'{self.ignore}/'
-            return path.realpath(frame.f_code.co_filename).startswith(
-                path.realpath(self.ignore)
-            )
+        return (getattr(self.ignore, MODULE_IGNORE_ID_NAME, None) ==
+                frame.f_globals.get(MODULE_IGNORE_ID_NAME, ''))
```

Comment:
```suggestion
        return (getattr(self.ignore, MODULE_IGNORE_ID_NAME, object()) ==
                frame.f_globals.get(MODULE_IGNORE_ID_NAME, object()))
```

Just in case somehow `None` or `''` is the actual value in one of those.